### PR TITLE
feat: support staging Cloud API endpoint

### PR DESCRIPTION
**Pull request by gee4vee**
Fix for https://github.ibm.com/org-ids/roadmap/issues/17379

### CODE REVIEW

File: .bluemix/toolchain.yml [hunk 1]
```diff
@@ -122,6 +122,12 @@ services:
             } else {
               'gitlab-pr-listener';
             }
+          IBMCLOUD_API: >
+            if ( $env.env_id === 'ibm:ys1:us-south') {
```
Comment (youming-lin) [inline .bluemix/toolchain.yml:126]: Should probably support `ibm:ys1:eu-gb` as well.

Comment (gee4vee) [reply .bluemix/toolchain.yml:126]: Good point. @maire-kehoe do we need to support all regions here?

Comment (gee4vee) [reply .bluemix/toolchain.yml:126]: Maybe we should have a contains clause and check for ys1 instead? Is that possible?

Comment (maire-kehoe) [reply .bluemix/toolchain.yml:126]: When I try to look at the toolchains in the `ibm:ys1:eu-gb` region, it shows me the same toolchain list that's in `ibm:ys1:us-south`, as in, it seems the staging London region is aliased to the same as the staging Dallas region for toolchain creation, and is not a real toolchains region. You can add support for it if you want, but I don't think that people test toolchains with that region very often.
Attempting to add a contains clause won't work as the `$env` expressions don't allow function calls.

Comment (youming-lin) [reply .bluemix/toolchain.yml:126]: Now that I think about it, shouldn't the API endpoint be predicated on the `registry-region` and `prod-region` values instead? After all the API you are targeting is affected by the region of the action you are trying to do (containerize or deploy to IKS).

**Comment by gee4vee** [comment]
@maire-kehoe @padraic-edwards So this is code complete and tested. I aded an `ibmcloud-api` param to all the tasks that take it in the pipelines. The CI pipeline almost succeeds except for `publish-deployable-task` which is failing with a 400. It is using the staging endpoint as expected. Is this expected? Is there a way to resolve this?

`Fail to create deployable mapping ! HTTP Status code = 400`

https://dev.console.test.cloud.ibm.com/devops/pipelines/tekton/2c070a7f-15f1-475f-9133-cfe6640543bc/runs/43cf8625-c76b-4fc4-acd1-6121bd9c5c38/publish-deployable-task/publish-deployable-mapping?env_id=ibm:ys1:us-south

File: .bluemix/toolchain.yml (hunk 1 above)
Comment (maire-kehoe) [reply .bluemix/toolchain.yml:126]: The code in question is doing an ibmcloud login to get a bearer token, using the bearer token to talk to the github-consolidated-broker for the current repo tool instance, to get the git token, to talk to the git server. So it needs to login using the same iam domain as the toolchain (ys1 vs yp), but the bearer token is region independent / global, so doesn't need to login to a specific region.
Code:
- https://github.com/open-toolchain/tekton-catalog/blob/ea3a38ca9f935d827d3ee3bf959b885746553158/git/task-clone-repo.yaml#L153

Some later code will log in again using the prod region when attempting to interact with the cluster.

So the current code should hopefully be ok.

**Review by maire-kehoe — APPROVED** [review]
LGTM. Thanks for the fix to the `publish-deployable-task`.

**Pull request merged**